packages/gafl-webapp-service/src/services/sales-api/sales-api-service.js

```diff
@@ -1,19 +1,77 @@
 import fetch from 'node-fetch'
 import { SALES_API_URL_DEFAULT, SALES_API_TIMEOUT_MS_DEFAULT } from '../../constants.js'
+import countryCodeProcessor from '../../processors/countries-helper.js'
 
 const urlBase = process.env.SALES_API_URL || SALES_API_URL_DEFAULT
 
+const headers = { 'Content-Type': 'application/json' }
+
 const fetchData = async url => {
+  let response
   try {
-    const response = await fetch(url.href, {
-      headers: { 'Content-Type': 'application/json' },
+    response = await fetch(url.href, {
+      headers,
       timeout: process.env.SALES_API_TIMEOUT_MS || SALES_API_TIMEOUT_MS_DEFAULT
     })
+  } catch (err) {
+    console.error('Unable to fetch data from the sales API service', err)
+    throw err
+  }
+
+  if (response.ok) {
     return response.json()
+  } else {
+    const mes = await response.json()
+    throw new Error(JSON.stringify(mes, null, 4))
+  }
+}
+
+const postData = async (url, payload) => {
+  let response
+  try {
+    response = await fetch(url.href, {
+      headers,
+      method: 'post',
+      body: JSON.stringify(payload),
+      timeout: process.env.SALES_API_TIMEOUT_MS || SALES_API_TIMEOUT_MS_DEFAULT
+    })
   } catch (err) {
-    console.error('Unable to connect to the sales API service', err)
+    console.error('Error posting data to the sales API service', err)
     throw err
   }
+
+  if (response.ok) {
+    return response.json()
+  } else {
+    const mes = await response.json()
```

Comment:
Should we console.error this here?

Would also be nice to prefix the json with an explanation that it is an unexpected error from the API.  Perhaps:
```const mes = `Unexpected response from Sales API ${await response.json()}```